todo_app/data/trello_items.py

```diff
@@ -35,34 +71,109 @@ def get_cards(id):
     return response
 
 
-def get_boardID(name):
-    searchUrl = baseUrl + "members/me/boards?fields=name"
+# get the lists on  board
+def get_boardLists(id):
+    searchUrl = f"{baseUrl}boards/{id}/lists"
 
     try:
-        response = requests.request("GET", searchUrl, params=query)
+        response = requests.request("GET", searchUrl, headers=headers, params=query)
         response.raise_for_status()
         response = response.json()
     except requests.exceptions.RequestException as exception:
         print(f"An error occured: {exception}")
         response = None
 
-    if response:
-        boardItem = next((obj for obj in response if obj["name"] == name), None)
-        if boardItem:
-            return boardItem["id"]
-        else:
-            return None
+    return response
+
+
+# get cards on a board
+def get_cards(id):
+    classItems = []
+    found = False
+
+    try:
+        # get all lists for the board
+        boardLists = get_boardLists(id)
+
+        # get the cards for each list
+        if len(boardLists) > 0:
+            for obj in boardLists:
+                cardsList = get_list_cards(obj["id"])
+                for cardList in cardsList:
+                    # if classItems is empty code will not execute so need an if
+                    for card in classItems:
+                        if cardList["id"] == card.id:
+                            found = True
+                            break
+                        else:
+                            found = False
+                    if found == False:
+                        date = datetime.strptime(
+                            cardList["dateLastActivity"], "%Y-%m-%dT%H:%M:%S.%fZ"
+                        )
```

Comment:
Looks like you handle the date in the constructor so we could always lose it here 😄